src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py

```diff
@@ -0,0 +1,230 @@
+        self.robot_model = self.robot.robot_model
+        self.reset()
+
+    def reset(self):
+        self._episode_start_pose = xyt2sophus(self.get_base_pose())
+        if self.visualizer is not None:
+            self.visualizer.reset()
+
+        # Switch control mode on the robot to nav
+        self.robot.switch_to_navigation_mode()
+        # put the robot in the correct mode with head facing forward
+        home_q = STRETCH_NAVIGATION_Q
+        # TODO: get this right
+        # tilted
+        home_q = self.robot_model.update_look_front(home_q.copy())
+        # Flat
+        # home_q = self.robot_model.update_look_ahead(home_q.copy())
+        self.goto(home_q, move_base=False, wait=True)
+
+    def apply_action(self, action: Action, info: Optional[Dict[str, Any]] = None):
+        """Discrete action space. make predictions for where the robot should go, move by a fixed
+        amount forward or rotationally."""
+        if self.visualizer is not None:
+            self.visualizer.visualize(**info)
+        continuous_action = np.zeros(3)
+        if action == DiscreteNavigationAction.MOVE_FORWARD:
+            print("FORWARD")
+            continuous_action[0] = self.forward_step
+        elif action == DiscreteNavigationAction.TURN_RIGHT:
+            print("TURN RIGHT")
+            continuous_action[2] = -self.rotate_step
+        elif action == DiscreteNavigationAction.TURN_LEFT:
+            print("TURN LEFT")
+            continuous_action[2] = self.rotate_step
+        else:
+            # Do nothing if "stop"
+            # continuous_action = None
+            # if not self.in_manipulation_mode():
+            #     self.switch_to_manipulation_mode()
+            pass
+
+        if continuous_action is not None:
+            if not self.in_navigation_mode():
+                self.robot.switch_to_navigation_mode()
+                rospy.sleep(self.msg_delay_t)
+            if not self.dry_run:
+                self.robot.nav.navigate_to(
+                    continuous_action, relative=True, blocking=True
+                )
+        rospy.sleep(0.5)
+
+    def get_observation(self) -> Observations:
+        """Get rgb/xyz/theta from this"""
+        rgb, depth = self.get_images(compute_xyz=False, rotate_images=True)
+        current_pose = xyt2sophus(self.get_base_pose())
+
+        # use sophus to get the relative translation
+        relative_pose = self._episode_start_pose.inverse() * current_pose
+        euler_angles = relative_pose.so3().log()
+        theta = euler_angles[-1]
+        # pos, vel, frc = self.get_joint_state()
+
+        # GPS in robot coordinates
+        gps = relative_pose.translation()[:2]
+
+        # Create the observation
+        obs = home_robot.core.interfaces.Observations(
+            rgb=rgb.copy(),
+            depth=depth.copy(),
+            gps=gps,
+            compass=np.array([theta]),
+            # base_pose=sophus2obs(relative_pose),
```

Comment:
should we clean up stuff like this that's commented out from the previous env?

Reply:
This is actually commented out even in the previous env :)